### PR TITLE
* Added cookbook for downlink between servers

### PR DESCRIPTION
**Pull request by PiggottDev**
Added a cookbook where a downlink is used between two servers.

A JoinMapLane is also created where the downlinks are on a separate server.

### CODE REVIEW

File: server_downlinks/src/main/java/swim/basic/BasicPlane.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        context.command("/supplier", "wakeup", Value.absent());
+
+        //Create a downlink from our join lane to log whenever stock changes in any warehouse
+        final MapDownlink<String, Integer> totalStock = context.downlinkMap()
+                .keyForm(Form.forString()).valueForm(Form.forInteger())
+                .nodeUri("/supplier").laneUri("stock")
+                .didUpdate(((key, newValue, oldValue) -> {
+                    System.out.println(key + " updated from " + oldValue + " to " + newValue);
+                }))
+                .open();
```
Comment (DobromirM) [inline server_downlinks/src/main/java/swim/basic/BasicPlane.java:46]: The did update method for the `JoinMapLane` can be moved to the lane itself in the `SupplierAgent`.

File: server_downlinks/src/main/java/swim/basic/CustomPlane.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    @Override
+    public void didStart() {
+        super.didStart();
+        // Immediately wake up WarehouseAgent upon plane load
+        context.command("/warehouse/cambridge", "wakeup", Value.absent());
+        //Add an item to the cambridge warehouse which will resupply immediately
+        context.command("/warehouse/cambridge", "addItem", Text.from("foo"));
```
Comment (DobromirM) [inline server_downlinks/src/main/java/swim/basic/CustomPlane.java:48]: Maybe add more warehouse endpoints.

**Review by DobromirM — CHANGES_REQUESTED** [review]
Looks good overall, but it might need to be simplified a bit because the `JoinMapLane` is adding some additional complexity.

**Comment by ajay-gov** [comment]
Will be good to showcase these 4 scenarios

- Downlink to a `ValueLane` in plane 1 from plane 2 using the  plane 2's context
- Downlink to a `MapLane` in plane 1 from plane 2 using the  plane 2's context
- Downlink to a `ValueLane` in plane 1 from an agent in plane 2 using the agent's context
- Downlink to a `MapLane` in plane 1 from an agent in plane 2 using the agent's context

**Comment by PiggottDev** [comment]
Resolved all of Dobromir's comments and added each scenario mentioned by Ajay

**Comment by ajay-gov** [comment]
@PiggottDev looks good! Couple of minor fixes and then it should be good to merge. 

It's hard to see where the log messages for the downlink callbacks are getting invoked i.e from the Plane or from the Agent. It might be good to make that explicit in the log messages. In general I think it is good to prefix the `nodeUri()` in the Agent log messages. So something like: `System.out.println(nodeUri() + ": .....")`

The Readme title still says JoinLanes,

**Comment by PiggottDev** [comment]
@ajay-gov Thanks - I've made the logging better and fixed the Readme title

**Comment by ajay-gov** [comment]
I'm ok if this gets merged into master if @DobromirM approves

**Review by DobromirM — APPROVED** [review]

**Pull request merged**